packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -256,6 +328,209 @@ function toCollectorInstrumentationLibrarySpans(
+  const valueType = metric.descriptor.valueType;
+  if (
+    metricKind === MetricKind.COUNTER ||
+    metricKind === MetricKind.SUM_OBSERVER
+  ) {
+    if (valueType === apiValueType.INT) {
+      return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_INT64;
+    }
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_DOUBLE;
+  } else if (metric.aggregator instanceof HistogramAggregator) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.HISTOGRAM;
+  } else if (valueType == apiValueType.INT) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.INT64;
+  } else if (valueType === apiValueType.DOUBLE) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.DOUBLE;
+  } else {
+    // TODO: Add Summary once implemented
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.INVALID_TYPE;
+  }
+}
+
+/**
+ * Given a MetricDescriptor, return its temporality in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorTemporality(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptorTemporality {
+  const metricKind = metric.descriptor.metricKind;
+  if (
+    metricKind === MetricKind.COUNTER ||
+    metricKind === MetricKind.SUM_OBSERVER
+  ) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.CUMULATIVE;
+  } else if (
+    metricKind === MetricKind.UP_DOWN_COUNTER ||
+    metricKind === MetricKind.UP_DOWN_SUM_OBSERVER
+  ) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  } else if (
+    metricKind === MetricKind.VALUE_OBSERVER ||
+    metricKind === MetricKind.VALUE_RECORDER
+  ) {
+    // TODO: Change once LastValueAggregator is implemented.
+    // If the aggregator is LastValue or Exact, then it will be instantaneous
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  }
+  return opentelemetryProto.metrics.v1.MetricDescriptorTemporality
+    .INVALID_TEMPORALITY;
+}
+
+/**
+ * Given a MetricRecord, return the Collector compatible type of MetricDescriptor
+ * @param metric
+ */
+export function toCollectorMetricDescriptor(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptor {
+  return {
+    name: metric.descriptor.name,
+    description: metric.descriptor.description,
+    unit: metric.descriptor.unit,
+    type: toCollectorType(metric),
+    temporality: toCollectorTemporality(metric),
+  };
+}
+
+/**
+ * Returns an Int64Point or DoublePoint to the collector
+ * @param metric
+ * @param startTime
+ */
+export function toSingularPoint(
+  metric: MetricRecord,
+  startTime: number
+): {
+  labels: opentelemetryProto.common.v1.StringKeyValue[];
+  startTimeUnixNano: number;
+  timeUnixNano: number;
+  value: number;
+} {
+  let pointValue: number;
+  if (metric.aggregator instanceof MinMaxLastSumCountAggregator) {
+    pointValue = (metric.aggregator.toPoint().value as Distribution).last;
```

Comment:
Why this cast? Type casts usually indicate to me that a case is not actually covered somewhere.

Reply:
I got this error: `Property 'last' does not exist on type 'number | Distribution | Histogram'` despite MinMaxLastSumAggregator returning a distribution.

It seems like this is done in [tests](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-metrics/test/Meter.test.ts#L1055) despite still guaranteeing a MinMaxLastSumAggregator